packages/segmented-control/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,314 @@
+
+  const [getRef, setRef] = useDynamicRefs<HTMLInputElement>();
+
+  const mode = darkMode ? 'dark' : 'light';
+
+  // If a value is given, then it's controlled
+  let isControlled = controlledValue != null ? true : false;
+
+  React.Children.forEach(children, (child, i) => {
+    if (isComponentType(child, 'SegmentedControlOption')) {
+      // If one of the options has been programatically checked, then it's controlled
+      if (child.props.checked != null) {
+        isControlled = true;
+      }
+
+      // If there's no default value given, then we fall back to the first option
+      if (!defaultValue && i == 0) {
+        defaultValue = child.props.value;
+      }
+    }
+  });
+
+  // Keep track of the uncontrolled value
+  const [uncontrolledValue, setUncontrolledValue] = useState<string>(
+    defaultValue,
+  );
+
+  const name = useIdAllocator({ prefix: 'radio-group', id: nameProp });
+
+  // Change handler
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (onChange) {
+      onChange(e);
+    }
+
+    if (!isControlled) {
+      setUncontrolledValue(e.target.value);
+    }
+  };
+
+  // Add internal props to children passed in
+  const renderedChildren = React.Children.map(children, (child, index) => {
+    if (!isComponentType(child, 'SegmentedControlOption')) {
+      console.warn(`${child} is not a SegmentedControlOption`);
+      return child;
+    }
+
+    const _checked: boolean = isControlled
+      ? child.props.value === controlledValue || !!child.props.checked
+      : child.props.value === uncontrolledValue;
+
+    return React.cloneElement(child, {
+      id: child.props.id || `${name}-${index}`,
+      darkMode,
+      size,
+      _checked,
+      _name: name,
+      _onChange: handleChange,
+      ref: setRef(`${name}-${index}`),
+    });
+  });
+
+  const selectedIndex = React.Children.toArray(
+    renderedChildren,
+  ).findIndex(child =>
+    isControlled
+      ? (child as React.ReactElement).props.value === controlledValue
+      : (child as React.ReactElement).props.value === uncontrolledValue,
+  );
+
+  /**
+   * Dynamically set the size & position of the selection indicator
+   */
+  const [selectionStyleDynamic, setSelectionStyleDynamic] = useState<string>();
+
+  // Update dynamic styles of the selection indicator
+  useEffect(() => {
+    const selectedElement = (getRef(
+      `${name}-${selectedIndex}`,
+    ) as React.RefObject<HTMLInputElement>).current;
+
+    if (selectedElement) {
+      const { offsetWidth: width, offsetLeft: left } = selectedElement;
+      setSelectionStyleDynamic(css`
+        width: ${width}px;
+        transform: translateX(${left}px);
+      `);
+    }
+  }, [selectedIndex, getRef, name, renderedChildren]);
+
+  /**
+   * TODO
+   * - remove hover focus ring
+   */
+
+  return (
+    <div className={cx(wrapperStyle, className)} {...rest}>
+      <Overline className={cx(labelStyle[mode])}>{label}</Overline>
```

Comment:
We shouldn't use `cx()` unless we're composing more than one className together.

Reply:
Also, `label` is optional, so we should conditionally render this component.